front/components/vaults/VaultCreateAppModal.tsx

```diff
@@ -121,7 +121,11 @@ export const VaultCreateAppModal = ({
                 name="name"
                 value={name}
                 onChange={(e) => {
-                  setName(e.target.value);
+                  setName(
+                    e.target.value
+                      .replace(/[^a-zA-Z0-9_-]/g, "")
```

Comment:
Isn't it suprising to change the value for the user magically?

Reply:
My main point is that it's a departure from what we do elsewhere esp as the Input component supports error message